controllers/alerts/secret.go

```diff
@@ -42,42 +49,41 @@ func (r *secretReconciler) EmptyObject() client.Object {
 	return &corev1.Secret{}
 }
 
+// UpdateExistingResource checks if the secret already exists and has the correct token.
+// If it does, it returns nil. If the secret exists but the token is incorrect, it deletes the old secret
+// and creates a new one with the updated token. If the secret does not exist, it creates a new one.
+// It deletes the old secret to force Prometheus to reload the configuration.
 func (r *secretReconciler) UpdateExistingResource(ctx context.Context, cl client.Client, resource client.Object, logger logr.Logger) (client.Object, bool, error) {
 	found := resource.(*corev1.Secret)
-	modified := false
 
 	token, err := authorization.CreateToken()
 	if err != nil {
 		return nil, false, err
 	}
 
-	if found.Data["token"] == nil || string(found.Data["token"]) != token {
-		found.StringData = map[string]string{
-			"token": token,
-		}
-		modified = true
+	// Check if the secret has the correct token
+	if found.Data["token"] != nil && string(found.Data["token"]) == token {
```

Comment:
It is safe to cast nil byte array to string. You can simplify a bit by
```suggestion
	if string(found.Data["token"]) == token {
```